compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/calls/tower/TowerLevels.kt

```diff
@@ -209,7 +209,7 @@ class ScopeTowerLevel(
             }
         }
         return when {
-            candidate !is FirBackingFieldSymbol -> null
+            candidate !is FirBackingFieldSymbol && candidate !is FirPropertySymbol -> null
```

Comment:
Actually, I tried to remove this condition at all (so that `Functions` can be treated in the same way), but it didn't have any differences (w.r.t. BB tests) but introduced many other diagnostic failures. So I guess there are other steps during the resolution that set/use dispatch receiver for function calls.

Reply:
Yet I can tell you it should not work this way. See comment above `ScopeTowerLevel` about dispatch receiver. I will investigate it to understand why it helps in some particular cases.